pkg/util/cluster/cluster.go

```diff
@@ -714,13 +728,28 @@ func (c *Cluster) deleteCluster(ctx context.Context, resourceGroup, clusterName
 	return nil
 }
 
-func (c *Cluster) deleteClusterResourceGroup(ctx context.Context, resourceGroup string) error {
+func (c *Cluster) ensureResourceGroupDeleted(ctx context.Context, resourceGroupName string) error {
```

Comment:
This was added due to some racey deletions and some stuff not getting cleaned up correctly in the new tenant. AFAIK in the past we were letting the VnetRG get cleaned up via the `purge:true` flag (https://github.com/Azure/ARO-RP/pull/3513/files @tsatam can you confirm this). This just makes all the deletion a bit more explicit but we still have the purge script as a fall back